event_sink_clickhouse/sinks/base_sink.py

```diff
@@ -62,3 +81,170 @@ def _send_clickhouse_request(self, request, expected_insert_rows=None):
             # ClickHouse can be configured not to return the metadata / summary we check above for
             # performance reasons. It's not critical, so we eat those here.
             return response
+
+
+class ModelBaseSink(BaseSink):
+    """
+    Base class for ClickHouse event sink, allows overwriting of default settings
+
+    This class is used for the model based event sink, which uses the Django ORM to write
+    events to ClickHouse.
+    """
+
+    unique_key = None
+    clickhouse_table_name = None
+    queryset = None
+    name = None
+    timestamp_field = None
+    serializer_class = None
+    model = None
+
+    def __init__(self, connection_overrides, log):
+        super().__init__(connection_overrides, log)
+
+        required_fields = [
+            self.model,
+            self.clickhouse_table_name,
+            self.timestamp_field,
+            self.unique_key,
+            self.name,
+        ]
+
+        if not all(required_fields):
+            raise NotImplementedError(
+                "ModelBaseSink needs to be subclassed with model, clickhouse_table_name,"
+                "timestamp_field, unique_key, and name"
+            )
+
+    def get_model(self):
```

Comment:
Could this be a property for the class?

```suggestion
     @property
    def model(self):
```

Reply:
Sure

Reply:
I was hoping that people could override the get_model function for some unusual cases like the case for XBlocks and XBlockRelationships, which are stored in MongoDB and there is no such ORM.